pynest/nest/tests/test_refractory.py

```diff
@@ -0,0 +1,271 @@
+    "iaf_psc_delta_canon",
+    "iaf_psc_exp_ps",
+]
+
+# models that cannot be tested
+ignore_model = [
+    "aeif_cond_alpha_RK5",  # this one is faulty and will be removed
+    "ginzburg_neuron",
+    "hh_cond_exp_traub",
+    "hh_psc_alpha",
+    "hh_psc_alpha_gap",
+    "iaf_chs_2007",
+    "iaf_chxk_2008",
+    "iaf_tum_2000",
+    "izhikevich",
+    "mcculloch_pitts_neuron",
+    "parrot_neuron",
+    "parrot_neuron_ps",
+    "pp_pop_psc_delta",
+    "pp_psc_delta",
+    "sli_neuron",
+]
+
+tested_models = [m for m in nest.Models("nodes") if (nest.GetDefaults(
+                 m, "element_type") == "neuron" and m not in ignore_model)]
+
+# additional parameters for the connector
+add_connect_param = {
+    "iaf_cond_alpha_mc": {"receptor_type": 7},
+}
+
+
+# --------------------------------------------------------------------------- #
+#  Simulation and refractory time
+# -------------------------
+#
+
+simtime = 100
+resolution = 0.1
+min_steps = 1
+max_steps = 200
+
+
+# --------------------------------------------------------------------------- #
+#  Test class
+# -------------------------
+#
+
+def foreach_neuron(func):
+    '''
+    Decorator that automatically does the test for all neurons.
+    '''
+    def wrapper(*args, **kwargs):
+        self = args[0]
+        msd = 123456
+        N_vp = nest.GetKernelStatus(['total_num_virtual_procs'])[0]
+        pyrngs = [np.random.RandomState(s) for s in range(msd, msd + N_vp)]
+        for name in tested_models:
+            nest.ResetKernel()
+            nest.SetKernelStatus({
+                'resolution': resolution, 'grng_seed': msd + N_vp,
+                'rng_seeds': range(msd + N_vp + 1, msd + 2 * N_vp + 1)})
+            func(self, name, **kwargs)
+    return wrapper
+
+
+class RefractoryTestCase(unittest.TestCase):
+    """
+    Check the correct implementation of refractory time in all neuronal models.
+    """
+
+    def compute_reftime(self, model, sd, vm, neuron):
+        '''
+        Compute the refractory time of the neuron.
+
+        Parameters
+        ----------
+        model : str
+          Name of the neuronal model.
+        sd : tuple
+            GID of the spike detector.
+        vm : tuple
+            GID of the voltmeter.
+        neuron : tuple
+            GID of the recorded neuron.
+
+        Returns
+        -------
+        t_ref_sim : double
+            Value of the simulated refractory period.
+        '''
+        spike_times = nest.GetStatus(sd, "events")[0]["times"]
+        if model in neurons_interspike:
+            # spike emitted at next timestep so substract resolution
+            return spike_times[1]-spike_times[0]-resolution
+        elif model in neurons_interspike_ps:
+            return spike_times[1]-spike_times[0]
+        else:
+            Vr = nest.GetStatus(neuron, "V_reset")[0]
+            times = nest.GetStatus(vm, "events")[0]["times"]
+            idx_max = (np.argwhere(np.isclose(times, spike_times[1]))[0][0]
```

Comment:
is there a reason why np.isclose is used here? Otherwise you could make this consistent with line 209.